src/osd/OSDMap.cc

```diff
@@ -3966,10 +3966,15 @@ void OSDMap::print_summary(Formatter *f, ostream& out,
 			   const string& prefix, bool extra) const
 {
   if (f) {
+    utime_t now = ceph_clock_now();
     f->dump_int("epoch", get_epoch());
     f->dump_int("num_osds", get_num_osds());
     f->dump_int("num_up_osds", get_num_up_osds());
+    f->dump_int("osd_up_since", (now - last_up_change).to_msec() / 1000);
```

Comment:
i'd prefer printing the information conveyed by osdmap itself which does not change overtime.

Reply:
Personally, I dont' understand what you wanted to say...is it bad or good, or what is wrong...

Reply:
@socketpair i think we should just print the timestamp of `last_up_change`.

Reply:
Yep, you are right. Agree.

Reply:
ok,changed.

Reply:
@tchaikov This pr last a little long ,pls have a look here.